applications/simcenter-designsafe-vm/tapisjob_app.sh

```diff
@@ -0,0 +1,67 @@
+#!/bin/bash
+
+handle_error() {
+  local EXITCODE=$1
+  echo "Phyiscs-based extractor job exited with an error status. $EXITCODE" >&2
+  echo "$EXITCODE" > "${_tapisExecSystemOutputDir}/tapisjob.exitcode"
+  exit $EXITCODE
+}
+
+set -x
+
+PBESCRIPT="${inputScript##*/}"
+echo "PBESCRIPT is $PBESCRIPT"
+
+inputPath="`pwd`/${inputDirectory}"
+INPUTDIR=${inputPath}
+if [ -z ${inputDirectory} ]; then
+    INPUTDIR="/corral/projects/NHERI/shared/${_tapisJobOwner}"
+fi
+
+#dataPath="`pwd`/${dataDirectory}"
+DATADIR_NAME="${dataDirectory##*/}"
+DATADIR=`python3 get_folder_mount_path.py $dataDirectory`
+
+DATADIR_MOUNT="--bind ${DATADIR}:/home/${_tapisJobOwner}/work/projects/$DATADIR_NAME"
+if [ -z ${dataDirectory} ]; then
+    projects_dir="$HOME/MyProjects"
+
+    mkdir -p "$projects_dir"
+    IFS=' ' read -r -a projects <<< "${_UserProjects}"
+    for project in "${projects[@]}"; do
+        IFS=',' read -r uuid projectId <<< "$project"
+        target_path="/corral/main/projects/NHERI/projects/$uuid"
+        symlink_path="$projects_dir/$projectId"
+
+        if [ -e "$target_path" ]; then
+            if [ ! -e "$symlink_path" ]; then
+                ln -s "$target_path" "$symlink_path"
+                echo "TACC: Project Links: Created symlink: $symlink_path -> $target_path"
+            else
+                echo "TACC: Project Links: Symlink already exists: $symlink_path"
+            fi
+        else
+            echo "TACC: Project Links: Target path does not exist: $target_path"
+        fi
+    done
+    DATADIR_MOUNT="--bind ${projects_dir}:/home/${_tapisJobOwner}/work/projects"
+fi
+
+echo "Running $PBESCRIPT"
+
+apptainer run \
+    --writable-tmpfs \
+    --memory 10G \
+    --bind $INPUTDIR:"/data/" \
+    --bind "/corral/main/projects/NHERI/shared/${_tapisJobOwner}":"/home/${_tapisJobOwner}/work/MyData" \
```

Comment:
Here MyData is under $HOME/work, but the rest are just under $HOME. Any reason not to include MyData and MyProjects just under $HOME and not $HOME/work? If copied from a jupyter app, $HOME/work is one of the jupyter default work dirs, so it need not necessarily apply here